components/ide/gha-update-image/index-jb-nightly.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+// Update JetBrains latest editor images
+//
+// ```
+// bun run index-jb-nightly.ts --task=<id> --productCode=<code>
+// ```
+
+import { $ } from "bun";
+import { appendGitHubOutput, pathToBackendPluginGradleLatest, readWorkspaceYaml } from "./lib/common";
+import { maybeCompatible, parseGradleProperties, parseGradlePropertiesFromTaskConfig } from "./lib/jb-helper/jb-helper";
+import { fetchProductReleases, ReleaseItem, releaseItemStr } from "./lib/jb-helper/jb-releases";
+import { getTaskFromArgs } from "./lib/jb-helper/jb-gradle-task-config";
+
+$.nothrow(); // git likes to respond with non-zero codes, but it is alright for us
+
+const task = getTaskFromArgs(true);
+
+if (task.id !== 1) {
+    throw new Error(`Only task 1 is supported, got ${task.id}`);
+}
+
+console.log(`Updating nightly editor for ${task.productId} (${task.productType})`);
+
+const { parsedObj: parsedWorkspaceYaml } = await readWorkspaceYaml();
+
+const downloadUrl = parsedWorkspaceYaml.defaultArgs[task.productId + "DownloadUrl"] as string;
+
+const latestGradle = parseGradleProperties(await Bun.file(pathToBackendPluginGradleLatest).text());
+
+const platformVersionType = "build";
+
+const releases = await fetchProductReleases({ productCode: task.productCode, productType: task.productType });
+
+let maybeCompatibleRelease: ReleaseItem | undefined;
+for (const release of releases) {
+    switch (platformVersionType) {
+        case "build": {
+            const ok = maybeCompatible(release, latestGradle);
```

Comment:
Major change: we compare `buildSince` field instead of (old logic) find a match value